source/common/tracing/dynamic_opentracing_driver_impl.cc

```diff
@@ -12,23 +12,15 @@ DynamicOpenTracingDriver::DynamicOpenTracingDriver(Stats::Store& stats, const st
   opentracing::expected<opentracing::DynamicTracingLibraryHandle> library_handle_maybe =
       opentracing::DynamicallyLoadTracingLibrary(library.c_str(), error_message);
   if (!library_handle_maybe) {
-    if (error_message.empty()) {
-      throw EnvoyException{fmt::format("{}", library_handle_maybe.error().message())};
-    } else {
-      throw EnvoyException{
-          fmt::format("{}: {}", library_handle_maybe.error().message(), error_message)};
-    }
+    throw EnvoyException{
+        fmt::format("{}: {}", library_handle_maybe.error().message(), error_message)};
   }
   library_handle_ = std::move(*library_handle_maybe);
 
   opentracing::expected<std::shared_ptr<opentracing::Tracer>> tracer_maybe =
       library_handle_.tracer_factory().MakeTracer(tracer_config.c_str(), error_message);
   if (!tracer_maybe) {
-    if (error_message.empty()) {
-      throw EnvoyException{fmt::format("{}", tracer_maybe.error().message())};
-    } else {
-      throw EnvoyException{fmt::format("{}: {}", tracer_maybe.error().message(), error_message)};
-    }
+    throw EnvoyException{fmt::format("{}: {}", tracer_maybe.error().message(), error_message)};
```

Comment:
I definitely thing this is easier to read than the prior code, but won't it be a bit confusing if error_message is empty?  Is that an unexpected corner case?

Reply:
It would have a extra trailing ":" if the error message is empty which might be a little confusing.

I could add something like a formatErrorMessage function that could do the conditional formatting and be tested directly if you think that's better?

Reply:
If it's likely to be empty I'd make it conditional.  Knowing little of opentracing I'm not sure if it's warranted. :-)

Reply:
Ok. Put the conditional formatting back in, but used a separate function so that everything has test coverage.